### PR TITLE
TRestEventRateAnalysisProcess: auto adjust under single/multi thread condition

### PR DESCRIPTION
**Pull request by nkx111**
nkx111 Ok: 31 [](https://gitlab.cern.ch/rest-for-physics/framework/-/commits/nkx111-patch-2) [](https://github.com/rest-for-physics/framework/commits/nkx111-patch-2)

The observable `SecondsFromStart` and `HoursFromStart` in `TRestEventRateAnalysisProcess` are not single thread observables. We can preserve multi-thread run for `TRestEventRateAnalysisProcess` if we don't need the rate observables. 

So this PR changes: if `TRestEventRateAnalysisProcess` detects that it is being running under multi-thread, it will turn off the observable `EventTimeDelay` and `MeanRate_InHz` automatically.

### CODE REVIEW

File: source/framework/analysis/inc/TRestEventRateAnalysisProcess.h
Comment (Vindaar) [inline source/framework/analysis/inc/TRestEventRateAnalysisProcess.h:61]: I'd avoid if expressions without curly brackets. Too easy to introduce bugs if the body is ever extended, especially because indentation (which is mixing tabs & spaces here btw) doesn't save you in C/C++.

File: source/framework/analysis/src/TRestEventRateAnalysisProcess.cxx
Comment (Vindaar) [inline source/framework/analysis/src/TRestEventRateAnalysisProcess.cxx:139]:
```suggestion
    // if is run under single thread mode, we add rate observables
    fRateAnalysis = GetNumberOfParallelProcesses() <= 1;
```

**Review by jgalan — APPROVED** [review]

**Comment by jgalan** [comment]
What about this PR? Should be approved or closed?

**Comment by nkx111** [comment]
still waiting for approval

File: source/framework/analysis/inc/TRestEventRateAnalysisProcess.h
Comment (Vindaar) [inline source/framework/analysis/inc/TRestEventRateAnalysisProcess.h:63]:
```suggestion
        if (fRateAnalysis) {
	    RESTMetadata << "Rate analysis is on under single thread mode" << RESTendl;
	}
	else {
	    RESTMetadata << "Rate analysis is off due to multi-thread" << RESTendl;
	}
```

**Review by Vindaar — APPROVED** [review]
Feel free to ignore the whitespace.

**Pull request merged**
